### PR TITLE
feat: reset encryption session (AR-2687)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2687" title="AR-2687" target="_blank">StoryAR-2687</a>  Reset encryption session
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Currently, we don't have the possibility to reset the encryption session.

### Solutions

Implement the User interaction mapping and the call to logic for the reset encryption session.
After we reset session, we should be able to decrypting messages again.

### Dependencies (Optional)

Needs to be merged after these kalium PRs:

https://github.com/wireapp/kalium/pull/1215
https://github.com/wireapp/kalium/pull/1222

Needs releases with:

- [x] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

### Attachments (Optional)

https://user-images.githubusercontent.com/5806454/206752449-d54ff4ec-23f5-4dc2-abcc-54534df8d80c.mov

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5152800)(5152800)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5152800)(5152800)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5152952)(5152952)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5152952)(5152952)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5154084)(5154084)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5154084)(5154084)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5154348)(5154348)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5154348)(5154348)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5154553)(5154553)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5154553)(5154553)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/9/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5157300)(5157300)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5157300)(5157300)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/12/) **failed**.

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/13/) **failed**.

**Comment by AndroidBob** [comment]
Build [14](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/14/) **failed**.

**Comment by AndroidBob** [comment]
Build [15](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/15/) **failed**.

**Comment by AndroidBob** [comment]
Build [16](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/16/) **failed**.

**Comment by AndroidBob** [comment]
Build [17](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/17/) **failed**.

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt [hunk 1]
```diff
@@ -842,9 +843,13 @@ class UseCaseModule {
...
     @Provides
-    fun providesIsSelfTeamMemberUseCase(
+    fun provideIsSelfTeamMemberUseCase(
         @KaliumCoreLogic coreLogic: CoreLogic,
         @CurrentAccount currentAccount: UserId
     ): IsSelfATeamMemberUseCase = coreLogic.getSessionScope(currentAccount).team.isSelfATeamMember
 
+    @ViewModelScoped
+    @Provides
+    fun provideResetSession(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: UserId): ResetSessionUseCase =
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt]: Just for consistency, can you rename it  to `provideResetSessionUseCase`

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt [hunk 2]
```diff
@@ -180,6 +187,20 @@ class ConversationMessagesViewModel @Inject constructor(
         }
     }
 
+    fun onResetSession(userId: UserId, clientId: String?) {
+        viewModelScope.launch {
+            conversationViewState =
+                when (withContext(dispatchers.io()) { resetSession(conversationId, userId, ClientId(clientId.orEmpty())) }) {
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt]: I think we don't need to switch contexts here, the use case should handle that ( I updated that in kalium PR)

**Review by ohassine — APPROVED** [review]
Nicely done 👌 just left tiny comments

File: app/src/test/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModelArrangement.kt [hunk 3]
```diff
@@ -96,6 +102,7 @@ class ConversationMessagesViewModelArrangement {
         coEvery { observeConversationDetails(any()) } returns flowOf()
         coEvery { getMessagesForConversationUseCase(any()) } returns messagesChannel.consumeAsFlow()
         coEvery { updateAssetMessageDownloadStatus(any(), any(), any()) } returns UpdateDownloadStatusResult.Success
+        coEvery { resetSession(any(), any(), any()) } returns Either.Right(Unit)
```
Comment (yamilmedina) [inline app/src/test/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModelArrangement.kt]: Why this mock fails 😠 ?

**Comment by AndroidBob** [comment]
Build [19](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/19/) **failed**.

**Comment by AndroidBob** [comment]
Build [20](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/20/) **failed**.

**Comment by AndroidBob** [comment]
Build [21](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/21/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5188700)(5188700)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5188700)(5188700)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [22](https://10.10.124.134/job/wire-android-reloaded/job/PR-1221/22/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5188952)(5188952)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5188952)(5188952)-dev-release.apk)

**Pull request merged**